hub/policy/src/main/java/uk/gov/ida/hub/policy/domain/controller/IdpSelector.java

```diff
@@ -18,6 +18,7 @@ private IdpSelector() {}
     public static IdpSelectedState buildIdpSelectedState(IdpSelectingState state,
```

Comment:
As with the class, I'm not a huge fan of the long signatures.
If you have any ideas that would be awesome.

Reply:
Normally you'd have some fields initialised when instantiating an object but this is a static method so it needs to have all of its parameters passed in. Unless it's worth/possible to inject the two proxy objects instead?

Reply:
Named parameters in other languages make this kind of thing much nicer.
You could use the builder pattern, but then you have to find a way of telling / ensuring that all the parameters have been passed in.

I think just leave it for now - I don't have a solution which is any less awkward that what it already was!